zilliqa/src/exec.rs

```diff
@@ -673,23 +676,101 @@ impl State {
         value: u128,
     ) -> Result<u64> {
         let gas_price = gas_price.unwrap_or(GAS_PRICE);
-        let gas = gas.unwrap_or(BLOCK_GAS_LIMIT);
 
+        let mut right = gas.unwrap_or(MAX_EVM_GAS_LIMIT).0;
```

Comment:
Maybe a nitpick, but can we save some type-safety by keeping this as an `EvmGas`? You may need to implement some extra traits to do the arithmetic, but I think that's reasonable.

Reply:
Yeah fair enough, I guess its completely local to this function.